htdocs/cron/class/cronjob.class.php

```diff
@@ -1518,6 +1520,33 @@ public function LibStatut($status, $mode = 0, $processing = 0, $lastresult = 0)
 
 		return dolGetStatus($this->labelStatus[$status], $this->labelStatusShort[$status], '', $statusType, $mode);
 	}
+
+	/**
+	 * Check if the lastoutput column can be set for the cronjob table.
+	 *
+	 * @param int $lastoutput_length The length of lastoutput.
+	 * @return bool True if lastoutput can be set, false otherwise.
+	 */
+	function canSetLastOutput($lastoutput_length) {
+		global $dolibarr_main_db_name;
+
+		$sql = "SELECT character_maximum_length
```

Comment:
Making external database access inside low level methods (like logging or for strong manipulation purpose) , above all on a external system database that may not exists or users may not have permission is forbidden.
You should be able to get the max length of a field into the ->fields array of the crud class. If field array not present, you can add it or use z hard coded value (lenght of field are hard coded int sql files, so it can also be hard coded into php files.